modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -254,6 +270,68 @@ TensorMechanicsAction::act()
   }
 }
 
+void
+TensorMechanicsAction::actOutputGeneration()
+{
+  // table data for output generation
+  struct OutputData
+  {
+    std::string _variable, _auxkernel, _tensor;
+    std::pair<unsigned int, unsigned int> _index;
+  };
+  const std::vector<OutputData> data = {
+      {"strain_xx", "RankTwoAux", "total_strain", {0, 0}},
+      {"strain_yy", "RankTwoAux", "total_strain", {1, 1}},
+      {"strain_zz", "RankTwoAux", "total_strain", {2, 2}},
+      {"strain_xy", "RankTwoAux", "total_strain", {0, 1}},
+      {"strain_yz", "RankTwoAux", "total_strain", {1, 2}},
+      {"strain_zx", "RankTwoAux", "total_strain", {2, 0}},
+      {"stress_xx", "RankTwoAux", "stress", {0, 0}},
+      {"stress_yy", "RankTwoAux", "stress", {1, 1}},
+      {"stress_zz", "RankTwoAux", "stress", {2, 2}},
+      {"stress_xy", "RankTwoAux", "stress", {0, 1}},
+      {"stress_yz", "RankTwoAux", "stress", {1, 2}},
+      {"stress_zx", "RankTwoAux", "stress", {2, 0}}};
+
+  //
+  // Add variables (optional)
+  //
+  if (_current_task == "add_variable" && getParam<bool>("add_variables"))
+  {
+    // Loop through output aux variables
+    for (auto out : _generate_output)
+    {
+      // Create displacement variables
+      _problem->addAuxVariable(data[out]._variable,
+                               FEType(Utility::string_to_enum<Order>("CONSTANT"),
+                                      Utility::string_to_enum<FEFamily>("MONOMIAL")),
```

Comment:
We can address this later, but it would be nice to have the option to use higher order monomial variables (same order as the displacement variables) for stress output.